### PR TITLE
Added `comment` configuration.

### PR DESCRIPTION
**Pull request by rozza**
Allows users to add a comment for debugging
and profiling purposes.

SPARK-377

### CODE REVIEW

File: src/integrationTest/java/com/mongodb/spark/sql/connector/mongodb/MongoSparkConnectorTestCase.java [hunk 1]
```diff
@@ -140,6 +157,59 @@ public void loadSampleData(
...
+              format("{profile: 2, filter: {ns: '%s.%s'}}", databaseName, collectionName)));
+
+      runnable.run();
+
+      Predicate<? super Document> commentsToIgnore =
+          d -> {
+            String profileComment = d.get("comment", "");
+            return d.containsKey("killCursors")
+                || profileComment.equals(IGNORE_COMMENT)
+                || profileComment.equals(TEST_COMMENT);
```
Comment (katcharov) [inline src/integrationTest/java/com/mongodb/spark/sql/connector/mongodb/MongoSparkConnectorTestCase.java:193]: Why are both "ignore" and "test" comments needed?

Comment (rozza) [reply src/integrationTest/java/com/mongodb/spark/sql/connector/mongodb/MongoSparkConnectorTestCase.java:193]: Good question: as both are excluded from the profiler logs, in one sense only one could be required.

However, in the tests it makes sense to differentiate between the configured comments (eg `TEST_COMMENT`) and any setup or assertion based commands that also generate profile logs.

I've updated the test code to filter out explicitly ignored commands and the searching of any other commands that are missing a comment. Hopefully, that will help clarify it.

```
      List<Document> profileDocs =
          profileCollection
              .find(
                  Filters.nor(Filters.exists("killCursors"), Filters.eq("comment", IGNORE_COMMENT)))
              .into(new ArrayList<>());

      List<String> withoutComment =
          profileDocs.stream()
              .filter(d -> !d.getEmbedded(asList("command", "comment"), "").equals(TEST_COMMENT))
              .map(Document::toJson)
              .collect(Collectors.toList());
```

**Review by katcharov — APPROVED** [review]
LGTM!

File: src/main/java/com/mongodb/spark/sql/connector/write/MongoDataWriter.java [hunk 2]
```diff
@@ -79,7 +79,8 @@ final class MongoDataWriter implements DataWriter<InternalRow> {
             schema, writeConfig.convertJson(), writeConfig.ignoreNullValues());
     this.writeConfig = writeConfig;
     this.epochId = epochId;
-    this.bulkWriteOptions = new BulkWriteOptions().ordered(writeConfig.isOrdered());
+    this.bulkWriteOptions =
+        new BulkWriteOptions().ordered(writeConfig.isOrdered()).comment(writeConfig.getComment());
```
Comment (vbabanin) [inline src/main/java/com/mongodb/spark/sql/connector/write/MongoDataWriter.java:83]: [nit] I noticed that in many places, the method chains are wrapped on a separate line, which I personally find easier to read

```suggestion
        new BulkWriteOptions()
        	.ordered(writeConfig.isOrdered())
       		.comment(writeConfig.getComment());
```

**Comment by vbabanin** [comment]
LGTM! Just a couple of nits

**Review by vbabanin — APPROVED** [review]

**Pull request merged**